app/src/test/java/com/example/mvvmcleanarchitecture/utils/LiveData+extensions.kt

```diff
@@ -0,0 +1,71 @@
+package com.example.mvvmcleanarchitecture.utils
+
+
+/*
+* Copyright (C) 2019 The Android Open Source Project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+/**
+ * Gets the value of a [LiveData] or waits for it to have one, with a timeout.
+ *
+ * Use this extension from host-side (JVM) tests. It's recommended to use it alongside
+ * `InstantTaskExecutorRule` or a similar mechanism to execute tasks synchronously.
+ */
+fun <T> LiveData<T>.getOrAwaitValue(
```

Comment:
I tu pytanie: skąd znajdujesz takie rozwiązanie open source np. napisane przez google ? 
W IDE można sobie ściągnąć przykładowae aplikacje opisujące dane uzycie, np. animacje, użycie kamery i często znajduję w nich rozwiązania od google które mi póżniej pomagają w swoich projektach. Czy w ten sposób znajdujesz takie rozwiązania czy dodatkowo poświęcasz swój czas na wertowanie wpisów na medium, twitter czy dokumentacji ? No i ogólnie szacun bo mnie zaskoczyłęś kilkukrotnie w swoim kursie takimi rzeczami 👍

Reply:
Akurat w znalezieniu tego konkretnego przykładu pomógł mi mój kolega po fachu :) Generalnie staram się być w miarę na bieżąco z tym co się dzieje w świecie Androida, więc czytam sobie artykuły na Medium albo przeglądam posty na Twitterze. Ale umówmy się, że bardzo dużo rozwiązań tego typu szuka się dopiero gdy ma się konkretny problem, bo wcześniej można nie widzieć potrzeby ich użycia. I tak było w tym przypadku, bo zaczęło się od tego, że nie działały mi testy, a zapomniałem, że w przypadku testowania trzeba mockować loopera (miałem już gdzieś w projektach fragment kodu, który za to odpowiadał, ale był napisany w inny sposób). No i tak trafiłem na to rozwiązanie :)